pytest_ansible/plugin.py

```diff
@@ -1,205 +1,153 @@
-        group.addoption('--ansible-become-method',
-                        action='store',
-                        dest='ansible_become_method',
-                        default=ansible.constants.DEFAULT_BECOME_METHOD,
-                        help="privilege escalation method to use (default: %%default), valid choices: [ %s ]" % (' | '.join(ansible.constants.BECOME_METHODS)))
-        group.addoption('--ansible-become-user',
-                        action='store',
-                        dest='ansible_become_user',
-                        default=ansible.constants.DEFAULT_BECOME_USER,
-                        help='run operations as this user (default: %default)')
+                    dest='ansible_become_user',
+                    default=ansible.constants.DEFAULT_BECOME_USER,
+                    help='run operations as this user (default: %(default)s)')
+    group.addoption('--ask-become-pass', '--ansible-ask-become-pass',
+                    action='store',
+                    dest='ansible_ask_become_pass',
+                    default=ansible.constants.DEFAULT_BECOME_ASK_PASS,
+                    help='ask for privilege escalation password (default: %(default)s)')
 
     # Add github marker to --help
     parser.addini("ansible", "Ansible integration", "args")
 
 
 def pytest_configure(config):
-    '''
-    Validate --ansible-* parameters.
-    '''
+    """Validate --ansible-* parameters."""
     log.debug("pytest_configure() called")
 
     config.addinivalue_line("markers", "ansible(**kwargs): Ansible integration")
 
     # Enable connection debugging
-    if config.getvalue('ansible_debug'):
-        if has_ansible_v2:
-            display.verbosity = 5
+    if config.option.verbose > 0:
+        if hasattr(ansible.utils, 'VERBOSITY'):
+            ansible.utils.VERBOSITY = int(config.option.verbose)
         else:
-            ansible.utils.VERBOSITY = 5
+            from ansible.utils.display import Display
+            display = Display()
+            display.verbosity = int(config.option.verbose)
 
     assert config.pluginmanager.register(PyTestAnsiblePlugin(config), "ansible")
 
 
-class PyTestAnsiblePlugin:
-
-    def __init__(self, config):
-        log.debug("PyTestAnsiblePlugin initialized")
-        self.config = config
-
-    def assert_required_ansible_parameters(self):
-        '''Helper method to assert whether the required --ansible-* parameters were
-        provided.
-        '''
+def pytest_generate_tests(metafunc):
+    """Generate tests when specific `ansible_*` fixtures are used by tests."""
+    log.debug("pytest_generate_tests() called")
 
-        errors = []
-
-        # Verify --ansible-host-pattern was provided
-        ansible_hostname = self.config.getvalue('ansible_host_pattern')
-        if ansible_hostname is None or ansible_hostname == '':
-            errors.append("Missing required parameter --ansible-host-pattern")
+    if 'ansible_host' in metafunc.fixturenames:
```

Comment:
On closer inspection, I'm wrong ... these changes do break that.  I'm debating fixing that after merge though.